src/types/receipt.ts

```diff
@@ -0,0 +1,121 @@
+// Types
+import { JsonRpcLog } from "./log.ts";
+
+// Starknet
+import { Event } from "../deps.ts";
+
+// Eth
+import {
+  bigIntToHex,
+  Bloom,
+  generateAddress,
+  hexToBytes,
+  JsonRpcTx,
+} from "../deps.ts";
+import { bytesToHex } from "../deps.ts";
+
+/**
+ * @param transaction A Ethereum transaction.
+ * @param logs A array of Ethereum logs.
+ * @param event The "transaction_executed" event.
+ * @param cumulativeGasUsed The cumulative gas used up to this transaction.
+ * @returns The Ethereum receipt.
+ */
+export function toEthReceipt(
+  { transaction, logs, event, cumulativeGasUsed }: {
+    transaction: JsonRpcTx;
+    logs: JsonRpcLog[];
+    event: Event;
+    cumulativeGasUsed: bigint;
+  },
+): JsonRpcReceipt | null {
+  // Gas used is the last piece of data in the transaction_executed event.
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const gasUsed = bigIntToHex(BigInt(event.data[event.data.length - 1]));
+  // Status is the second to last piece of data in the transaction_executed event.
+  // https://github.com/kkrt-labs/kakarot/blob/main/src/kakarot/accounts/eoa/library.cairo
+  const status = bigIntToHex(BigInt(event.data[event.data.length - 2]));
+  const contractAddress = transaction.to === null
+    ? bytesToHex(generateAddress(
+      hexToBytes(transaction.from),
+      hexToBytes(transaction.nonce),
+    ))
+    : null;
+
+  return {
+    transactionHash: transaction.hash,
+    transactionIndex: transaction.transactionIndex,
+    blockHash: transaction.blockHash,
+    blockNumber: transaction.blockNumber,
+    from: transaction.from,
+    to: transaction.to,
+    cumulativeGasUsed: bigIntToHex(cumulativeGasUsed),
+    gasUsed,
+    effectiveGasPrice: transaction.gasPrice,
+    // Incorrect, should be as in EIP1559
+    // min(transaction.max_priority_fee_per_gas, transaction.max_fee_per_gas - block.base_fee_per_gas)
+    // effective_gas_price = priority_fee_per_gas + block.base_fee_per_gas
+    contractAddress: contractAddress,
+    logs,
+    logsBloom: logsBloom(logs.map(toLog)),
+    status,
+  };
+}
+
+type Log = [address: Uint8Array, topics: Uint8Array[], data: Uint8Array];
+
+/**
+ * @param log A Ethereum json rpc log.
```

Comment:
i think it's not so clearly written idk it's hard to read

Reply:
I'm not sure what you mean, the comment is difficult to read?

Reply:
yes

Reply:
```suggestion
 * @param log - JSON RPC formatted Ethereum Log.
```